lib/browser/aws_iot.ts

```diff
@@ -201,21 +203,31 @@ export class AwsIotMqttConnectionConfigBuilder {
     }
 
     /**
-     * Configures AWS credentials (usually from Cognito) for this connection
+     * Configures Static AWS credentials for this connection. 
+     * Please note that the static credential will fail when the web session expired.
      * @param aws_region The service region to connect to
      * @param aws_access_id IAM Access ID
      * @param aws_secret_key IAM Secret Key
      * @param aws_sts_token session credentials token (optional)
-     *
+     * 
      * @returns this builder object
      */
-    with_credentials(aws_region: string, aws_access_id: string, aws_secret_key: string, aws_sts_token?: string) {
-        this.params.credentials = {
-            aws_region: aws_region,
-            aws_access_id: aws_access_id,
-            aws_secret_key: aws_secret_key,
-            aws_sts_token: aws_sts_token,
-        };
+     with_credentials(aws_region: string, aws_access_id: string, aws_secret_key: string, aws_sts_token?: string) {
+        const options = new StaticCredentialOptions(aws_region, aws_access_id, aws_secret_key,aws_sts_token);
+        const provider = new StaticCredentialProvider(options);
+        this.params.credentials_provider = provider;
+        return this;
+    }
+
+    /**
+     * Configures credentials provider (currently support for AWS Cognito Credential Provider) for this connection
+     * @param customer_provider credential provider used to update credential when session expired (optional)
+     * 
+     * @returns this builder object
+     */
+    with_credential_provider( customer_provider : CredentialsProvider) {
+        this.params.credentials_provider = customer_provider;
+        setInterval(()=>{ customer_provider.refreshCredential();}, customer_provider.expire_interval_in_ms);
```

Comment:
Mirroring the above comment, I think it's best if we left this up to the provider subclass.  As it currently stands, every connection you use the provider with will spawn an independent refresher task which isn't what we want.